pelicun/uq.py

```diff
@@ -1453,15 +1534,15 @@ def inverse_transform(self, values: np.ndarray) -> np.ndarray:
           too small
 
         """
-        mu, sig = self.theta[:2]
+        theta, truncation_limits = self._prepare_arrays(values)
+        mu, sig = theta.T
 
         if np.any(~np.isnan(self.truncation_limits)):
-            a, b = self.truncation_limits
+            a, b = truncation_limits.T
 
-            if np.isnan(a):
-                a = -np.inf
-            if np.isnan(b):
-                b = np.inf
+            # Replace NaN values
+            a = np.nan_to_num(a, nan=-np.inf)
+            b = np.nan_to_num(b, nan=np.inf)
 
             p_a, p_b = (norm.cdf((lim - mu) / sig) for lim in (a, b))
 
```

Comment:
There is a check below (I can't add a comment there) that makes sure `p_b-p_a > 0`. I suggest having a similar check above in the `cdf` function, especially because a p_b-p_a = 0 there would lead to dividing by zero.
And the same applies to other functions, so maybe we could have a private function that checks these probabilities and we just call that from everywhere.